include/fmt/format.h

```diff
@@ -553,6 +553,7 @@ FMT_CONSTEXPR auto fill_n(OutputIt out, Size count, const T& value)
 template <typename T, typename Size>
 FMT_CONSTEXPR20 auto fill_n(T* out, Size count, char value) -> T* {
   if (is_constant_evaluated()) return fill_n<T*, Size, T>(out, count, value);
+  static_assert(1 ==  sizeof(T), "sizeof(T) must be 1 to use char for initialization");
```

Comment:
nit: extra space after `==` and normally the literal is on the right-hand side: `sizeof(T) == 1`.